mantidproject/mantid
# Algorithms G Updated with HistogramData

Refactored files to use HistogramData. 
- GeneralisedSecondDifference.cpp
- GenerateEventsFilter.cpp
- GeneratePeaks.cpp
- GetAllEi.cpp
- GetDetectorOffsets.cpp
- GetDetOffsetsMultiPeaks.cpp
- GetEi.cpp
- GetEi2.cpp
- GetTimeSeriesLogInformation.cpp
- GravitySANSHelper.cpp
- GroupWorkspaces.cpp

Added performance tests for:
- GeneralisedSecondDifference
- GeneralisedEventsFilter
- GeneratePeaks
- GetAllEi
- GetDetectorOffsets
- GetEi
- GetTimeSeriesLogInformation
- GetDetOffsetsMultiPeaks

**To test:**

Run the Algorithms, check that the output is the same as before refactoring, functionality should not be changed.



Fixes #16959.


## _Does not need to be in the release notes._
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
